test.js

```diff
@@ -137,3 +137,19 @@ describe('runAsync', function () {
     }, /No Promise Implementation/);
   });
 });
+
+describe('runAsync.cb', function () {
+  it('handles callback parameter', function (done) {
```

Comment:
From #14 it seems you want to support returning Promises as well, I suspect that is going to result in `args.length === cb.length - 1`, which means we will ignore the Promise return value.

I think you should add a test for a Promise returning user function (I think it's going to be a problem).